hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/RestResources.java

```diff
@@ -563,6 +564,117 @@ public Response associateMetrics(@PathParam("environmentId") String environmentI
+                resourceSingle.configuration().create(configuration);
+                break;
+            case connectionConfiguration:
+                resourceSingle.connectionConfiguration().create(configuration);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported data type: " + dataType.toString());
+        }
+        return Response.noContent().build();
+    }
+
+    @GET
+    @javax.ws.rs.Path("/{environmentId}/{feedId}/resources/{resourcePath:.+}/data")
+    @ApiOperation("Retrieves the configuration of a resource")
+    @ApiResponses({
+                          @ApiResponse(code = 204, message = "OK"),
+                          @ApiResponse(code = 404, message = "Tenant, environment, resource or feed doesn't exist",
+                                       response = ApiError.class),
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response getConfiguration(@PathParam("environmentId") String environmentId,
+                                     @PathParam("feedId") String feedId,
+                                     @PathParam("resourcePath") String resourcePath,
+                                     @DefaultValue("configuration") @QueryParam("dataType") DataEntity.Role dataType) {
+
+        return getConfigurationHelper(environmentId, feedId, resourcePath, dataType);
+    }
+
+    @GET
+    @javax.ws.rs.Path("/{environmentId}/resources/{resourcePath:.+}/data")
+    @ApiOperation("Retrieves the configuration of a resource")
+    @ApiResponses({
+                          @ApiResponse(code = 204, message = "OK"),
+                          @ApiResponse(code = 404, message = "Tenant, environment or resource doesn't exist",
+                                       response = ApiError.class),
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response getConfiguration(@PathParam("environmentId") String environmentId,
+                                     @PathParam("resourcePath") String resourcePath,
+                                     @DefaultValue("configuration") @QueryParam("dataType") DataEntity.Role dataType) {
+
+        return getConfigurationHelper(environmentId, null, resourcePath, dataType);
+    }
+
+    private Response getConfigurationHelper(String environmentId, String feedId, String resourcePath,
+                                               DataEntity.Role dataType) {
+        String tenantId = getTenantId();
+        CanonicalPath resource = composeCanonicalPath(tenantId, environmentId, feedId, resourcePath);
+
+        Resources.Single resourceSingle = inventory.inspect(resource, Resources.Single.class);
+        DataEntity entity;
+        switch (dataType) {
+            case configuration:
+                entity = resourceSingle.configuration().get(null).entity();
```

Comment:
+1 this is a good start and we can expand on it later. The backend support is kinda there so it'd only be a matter of changing the REST API.